### PR TITLE
bit corruption env and metadata block changes

### PR DESCRIPTION
**Pull request by amytai**
Env that injects bit corruptions as well as hardening metadata blocks in SST files. Metadata block layout is detailed in a separate internal Quip document.

### CODE REVIEW

**Comment by siying** [comment]
Your new test failed in Travis.

File: table/block_based_table_builder.cc [hunk 1]
```diff
@@ -650,16 +650,26 @@ Status BlockBasedTableBuilder::Finish() {
...
+      filter_blocks.push_back(filter_content);
     }
   }
+  // END write filter block
+
+  // Locally replicate filter block(s) again.
+  BlockHandle filter_block_handle2;
+  for (std::vector<Slice>::iterator it = filter_blocks.begin();
+        it != filter_blocks.end(); it++) {
+    WriteRawBlock(*it, kNoCompression, &filter_block_handle2);
```
Comment (siying) [inline table/block_based_table_builder.cc]: Can you guard it as an option?

Comment (amytai) [reply table/block_based_table_builder.cc]: Guarded everything with BlockBasedTableOption::double_metadata

**Comment by siying** [comment]
I checked with @ajkr who said it's probably never means to be committed. So I guess we can close it. By the way, congratulations to @amytai and @ajkr on the ATC paper using this PR in the experiment:p

**Pull request closed without merging**